client/src/main/java/com/aliyun/emr/rss/client/read/RetryingChunkClient.java

```diff
@@ -217,7 +217,10 @@ private synchronized boolean shouldRetry(Throwable e) {
             || e instanceof TimeoutException
             || (e.getCause() != null && e.getCause() instanceof TimeoutException)
             || (e.getCause() != null && e.getCause() instanceof IOException);
-    return isIOException && hasRemainingRetries();
+    boolean isSortException =
+        e instanceof RuntimeException
+            && e.getMessage().startsWith("com.aliyun.emr.rss.service.deploy.worker.exception");
+    return (isIOException || isSortException) && hasRemainingRetries();
```

Comment:
I am not sure if it's ok to do like this, but at least, here miss to handle `openStream` return `null`.

And we can avoid return null and show clear error message for each case

Reply:
I think this section has some problems. SortedAccessHDFSException is not a subclass of RuntimeException. So this section cannot work properly.

Reply:
> I think this section has some problems. SortedAccessHDFSException is not a subclass of RuntimeException. So this section cannot work properly.

When exception pass to Client side, it had been wrapped as a RPCFailure
```
new RpcFailure(
          request.requestId,
          Throwables.getStackTraceAsString(fnf))
```